plugins/available/extract.plugin.bash

```diff
@@ -37,7 +37,7 @@ End-Of-Usage
 
         local -r filename=$(basename -- $1)
         local -r filedirname=$(dirname -- $1)
-        local targetdirname=$(sed 's/\(\.tar\.bz2$\|\.tbz$\|\.tbz2$\|\.tar\.gz$\|\.tgz$\|\.tar$\|\.tar\.xz$\|\.txz$\|\.tar\.Z$\|\.7z$\)//g' <<< $filename)
+        local targetdirname=$(sed 's/\(\.tar\.bz2$\|\.tbz$\|\.tbz2$\|\.tar\.gz$\|\.tgz$\|\.tar$\|\.tar\.xz$\|\.txz$\|\.tar\.Z$\|\.7z$\|\.nupkg$\|\.zip$\|\.war$\|\.jar$\)//g' <<< $filename)
         if [ "$filename" = "$targetdirname" ]; then
             # archive type either not supported or it doesn't need dir creation
             targetdirname=""
```

Comment:
I have a question why should `targetdirname` be set to empty string. I think we can comment this line.

Reply:
I think you are correct, and we transform this check into 
```bash
if [ "$filename" != "$targetdirname" ]; then
    # need dir creation
    mkdir -p -v "$targetdirname"
fi
```

Nice catch!